### PR TITLE
Add support for background image detection for c.tti.hi

### LINKED ISSUE #237: c_tti_hi should support background images
Currently turning on hero-image detection to measure c_tti_hi works only on `img` elements, however we can make it detect css `background-image` on `div` elements as well. I can open a PR if that sounds interesting.

### PR DESCRIPTION
**Pull request by hakimelek**
Adding the possibility to detect background-image urls on elements containing the hero image selector. 

Example:

```
<div class="hero-image" />

div {
   background: url('/image.png'); 
}
```

This should resolve #237.

### CODE REVIEW

File: plugins/continuity.js [hunk 1]
```diff
@@ -1382,6 +1382,13 @@
 					src = elements[i].currentSrc ||
 						elements[i].src ||
 						(typeof elements[i].getAttribute === "function" && elements[i].getAttribute("xlink:href"));
+	
+					// if src if not defined, look for it in css background image
+					if (!src) {
+						var bgStyle = BOOMR.window.getComputedStyle(elements[i]).getPropertyValue('background');
+						var bgImgUrl = bgStyle.match(/url\(["']?([^"']*)["']?\)/)[1];
```
Comment (bluesmoon) [inline plugins/continuity.js]: This will throw an error if `match` returns `null`.

Comment (hakimelek) [reply plugins/continuity.js]: Oups! forgot about the array while testing, just pushed a fix.

File: plugins/continuity.js [hunk 2]
```diff
@@ -1382,6 +1382,15 @@
 					src = elements[i].currentSrc ||
 						elements[i].src ||
 						(typeof elements[i].getAttribute === "function" && elements[i].getAttribute("xlink:href"));
+	
+					// if src if not defined, look for it in css background image
+					if (!src) {
+						var bgStyle = BOOMR.window.getComputedStyle(elements[i]).getPropertyValue('background');
```
Comment (nicjansma) [inline plugins/continuity.js]: I think we'll need a `typeof BOOMR.window.getComputedStyle === "function"` check as well, as that's only supported in IE9+ and Edge12+

Comment (hakimelek) [reply plugins/continuity.js]: Sure, I will do that and give the test a shot.

**Comment by nicjansma** [comment]
@hakimelek will you still be able to work on the changes for this?

Thanks for your help!

**Comment by hakimelek** [comment]
Hey @nicjansma, I pushed a new change to account for "getComputedStyle" support. For the test I was unsure how to debug it. I added a new <div> with a background image. I couldn't find a way to break the test even if I remove all the html tag. I was running `grunt test:debug`.  Sorry if this took longer than expected.

**Comment by nicjansma** [comment]
`17-tti-hero-images` is a good one to start with!  Here's what I'd recommend:

1. Copy `17-tti-hero-images.js` and `.html` to something like `33-tti-hero-images-bg-image.js` and `.html`
2. Replace references of `17-tti-hero-images` in both to `33-tti-hero-images-bg-image`
2. Change all of the ``, without your change, it should fail because it's not waiting on them.

**Comment by hakimelek** [comment]
@nicjansma Thanks for your patience, I added a test like you suggested. The only difference from the initial "hero-image" is that I am creating a new `Img` element for the same `src` defined in the background image since the `onload` function is no longer available.

File: plugins/continuity.js [hunk 3]
```diff
@@ -1383,6 +1383,17 @@
 						elements[i].src ||
 						(typeof elements[i].getAttribute === "function" && elements[i].getAttribute("xlink:href"));
 
+					// if src if not defined, look for it in css background image
+					if (!src) {
+						if (typeof BOOMR.window.getComputedStyle === "function") {
+							var bgStyle = BOOMR.window.getComputedStyle(elements[i]).getPropertyValue('background');
```
Comment (nicjansma) [inline plugins/continuity.js]: Last thing, can we do `null` checks on the return values of both `getComputedStyle()` and `getPropertyValue()`?

I see some [random notes about them returning `null`](https://www.fxsitecompat.dev/en-CA/docs/2018/getcomputedstyle-no-longer-returns-null-when-style-can-t-be-retrieved/) in [some cases](http://forums.mozillazine.org/viewtopic.php?t=647310) on the web.

Also, please run `grunt lint` -- there's a few small lint fixes needed.

That should be it!  Test looks great, thanks.

Comment (hakimelek) [reply plugins/continuity.js]: Changes made.

File: tests/page-templates/18-usertiming/06-usertiming-no-compression.js [hunk 4]
```diff
@@ -15,7 +15,7 @@ describe("e2e/18-usertiming/06-usertiming-no-compression", function() {
 			assert.isString(b.usertiming);
 			var data = JSON.parse(b.usertiming);
 			assert.isTrue("mark" in data);
-			var marks = data.mark
+			var marks = data.mark;
```
Comment (hakimelek) [inline tests/page-templates/18-usertiming/06-usertiming-no-compression.js]: Caught by linter.

**Comment by nicjansma** [comment]
Looks great thanks!

**Pull request merged**